src/theory/datatypes/datatypes_sygus.h

```diff
@@ -117,29 +124,140 @@ class SygusSymBreakNew
     /** For each term, whether this cache has processed that term */
     std::map< Node, bool > d_search_val_proc;
   };
-  // anchor -> cache
+  /** An instance of the above cache, for each anchor */
   std::map< Node, SearchCache > d_cache;
   /** a sygus sampler object for each (anchor, sygus type) pair
    *
    * This is used for the sygusRewVerify() option to verify the correctness of
    * the rewriter.
    */
   std::map<Node, std::map<TypeNode, quantifiers::SygusSampler>> d_sampler;
-  Node d_null;
+  /** Assert tester internal
+   *
+   * This function is called when the tester with index tindex is asserted for
+   * n, exp is the tester predicate. For example, for grammar:
+   *   A -> A+A | x | 1 | 0
+   * when is_+( d ) is asserted, assertTesterInternal(0, d, is_+( d ),...) is
+   * called. This function may add lemmas to lemmas, which are sent out on the
+   * output channel of datatypes by the caller.
+   */
   void assertTesterInternal( int tindex, TNode n, Node exp, std::vector< Node >& lemmas );
-  // register search term
+  /**
+   * This function is called when term n is registered to the theory of
+   * datatypes. It makes the appropriate call to registerSearchTerm below,
+   * if applicable.
+   */
+  void registerTerm(Node n, std::vector<Node>& lemmas);
+  /** Register search term
+   *
+   * This function is called when selector chain S_1( ... S_m( n ) ... ) is
+   * registered to the theory of datatypes, where tn is the type of n,
+   * d indicates the depth of n (the sum of weights of the selectors S_1...S_m),
+   * and topLevel is whether n is a top-level term (see d_is_top_level).
+   *
+   * The purpose of this function is to notify this class that symmetry breaking
+   * lemmas should be instantiated for n. Any symmetry breaking lemmas that
+   * should be immediately applied for n are added to lemmas in this call.
```

Comment:
Clarify the difference between symmetry breaking lemmas that "immediately apply" to those that do not.